perspective.el

```diff
@@ -950,6 +960,209 @@ perspective beginning with the given letter."
+Frames and their associated perspectives will also be saved
+(but not the original frame sizes).
+
+Buffers with * characters in their names, as well as buffers without
+associated files will be ignored. If such buffers are currently
+visible in a perspective as windows, they will be saved as
+'*scratch* (persp)' buffers."
+  (interactive (list
+                (read-file-name "Save perspective state to file: "
+                                persp-state-default-file
+                                persp-state-default-file)
+                t))
+  (unless persp-mode
+    (error "persp-mode not enabled, nothing to save"))
+  (lexical-let ((target-file (if (and file (not (string-equal "" file)))
+                                 ;; file provided as argument, just use it
+                                 (expand-file-name file)
+                               ;; no file provided as argument
+                               (if interactive?
+                                   ;; return nil in interactive call mode, since
+                                   ;; read-file-name should have provided a reasonable
+                                   ;; default
+                                   nil
+                                 ;; in non-interactive call mode, we want to fall back to
+                                 ;; the default, but only if it is set
+                                 (if (and persp-state-default-file
+                                          (not (string-equal "" persp-state-default-file)))
+                                     (expand-file-name persp-state-default-file)
+                                   nil)))))
+    (unless target-file
+      (error "No target file specified"))
+    ;; overwrite the target file if:
+    ;; - the file does not exist, or
+    ;; - the file is not the one set in persp-state-default-file, or
+    ;; - the user called this function with a prefix argument, or
+    ;; - the user approves overwriting the file when prompted
+    (when (and (file-exists-p target-file)
+               (not (string-equal (if (and persp-state-default-file
+                                           (not (string-equal "" persp-state-default-file)))
+                                      (expand-file-name persp-state-default-file)
+                                    "")
+                                  target-file))
+               (not (or current-prefix-arg
+                        (yes-or-no-p "Target file exists. Overwrite? "))))
+      (error "persp-state-save cancelled"))
+    ;; actually save
+    (persp-save)
+    (lexical-let ((state-complete (make-persp--state-complete
+                                   :files (persp--state-file-data)
+                                   :frames (persp--state-frame-data))))
+      (when (file-exists-p target-file)
+        (delete-file target-file))
+      (append-to-file (prin1-to-string state-complete) nil target-file))))
+
+;;;###autoload
+(defun persp-state-load (file)
```

Comment:
Should the `file` parameter be made `&optional`, to support using `persp-state-default-file` as the default/fall-back value?

Reply:
It already defaults to `persp-state-default-file` in interactive mode. It can be made optional in non-interactive mode, but that would require pulling in all the extra dispatch machinery from `persp-state-save`. I guess it would be useful in `after-init-hook` to restore sessions after Emacs starts, but I personally wouldn't feel comfortable with doing that automatically (and it's not like someone might forget to restore the session, unlike at shutdown).